### DISCUSSION TITLE
Subtitle

### DISCUSSION THREAD

**Discussion by imtrobin** [aclap-dev/video-downloadhelper#1204] [Help and Q&A]
### Your question

Hi, I'm trying to download a video that uses multiple VTT files (IDM shows this). Is this able to download or combine them into a SRT?

### Addon & CoApp version and details

version: 9.1.0.43
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36
{
    "http_media_download_strategy": "inbrowser",
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "open_count_store": 19,
    "successfull_dl": 4,

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

We have been clamoring for subtitle support in VDH for years.  @paulrouget Paul has said this feature is definitely going to be in VDH.  But we have no estimate on when that will happen.  In the meantime, you are left to analyze the Developer Tools & find whatever captions might exist that way.  Then you have several ways to get the subtitles, among them ffmpeg.

**Reply by GrampaWildWilly** [reply]
If you give me an example video, I'll try to show you how to get the captions.

**Reply by imtrobin** [reply]
Thanks, I can get the captions with IDM, but it's multiple VTT files which I hoping there is a tool to merge/fix them

**Reply by GrampaWildWilly** [reply]
What do you mean by multiple VTT files?  Is each VTT for the full duration of the video?  This would make sense if each VTT were of a different language.  Or is the subtitle file chopped up into segments  that, when placed end to end, would cover the full duration of the video?  This sounds interesting.  I'd still like you to give me the URL.

**Reply by imtrobin** [reply]
The link is a private paid course, so can't share sorry. There are multiple VTT files, each a few seconds, same language.

**Reply by GrampaWildWilly** [reply]
That's too bad.

And only a few seconds?  That is nasty.  Not the way it's supposed to be.  I've never encountered such a thing.  That seems like somebody didn't understand how things are supposed to be.  If they are actually meant to be placed end to end, then you can weld them together with ffmpeg.  The ffmpeg command would be this:

ffmpeg -f concat -safe 0 -i "list file name" -codec: copy "output VTT file name" 2>"ffmpeg log file name"

There's 3 file names there.  I'm showing them in quotation marks.  This is a safety measure to account for spaces & other odd things in file names.

The list file contains lines that look like this:

file 'VTT fragment file name x'

The first token on each line in the list file must be the word `file`.  Here, the file name must be surrounded by apostrophes.  This is a stupid restriction of the ffmpeg concat function.  You would have one line in the list file for each fragment of subtitles.  Be sure you get them in the right sequence.  I would choose file names that contain a sequence number to help you with that.

The output file name would be in whatever directory you choose, but the file extension should be .vtt.

The 2> bit at the end is standard Windows redirection.  It captures the output of ffmpeg.  If you don't code that, the output goes whizzing by in the command window faster than you can read it.  So even though it's not entirely necessary, you should do that.  If anything goes wrong, you will be able to read that file & debug whatever happened.

Let me know how this goes.

**Reply by imtrobin** [reply]
This is an example of the VTT file, u better believe it, just 3 item in one VTT file, there are multiples of them.

WEBVTT
X-TIMESTAMP-MAP=MPEGTS:900000,LOCAL:00:00:00.000

YYYYY
00:00:06.000 --> 00:00:08.180
XXX

YYYYY
00:00:08.190 --> 00:00:10.840
XXX

YYYYY
00:00:11.090 --> 00:00:12.000
XXX

**Reply by GrampaWildWilly** [reply]
What are the things you are showing as YYYYY?  Are those sequence numbers?  In that case, these are not VTT files, all other things saying they are.  They are SRT files.  The sequence numbers distinguish SRT from VTT.  Every subtitle in an SRT file has a sequence number.  To my way of thinking, this makes SRT an inferior format to VTT.  True VTT has no sequence numbers.  Not only does the absence of sequence numbers make VTT much easier to manipulate, but typically when you compare a VTT file & an SRT file of the same captions, the VTT is at least 1/3 smaller than the SRT.

So, despite somebody on your web site clearly not having a proper grasp of what subtitles are supposed to look like, were you able to weld the files together with ffmpeg using the approach I described?  You would, of course, have to name the output file something.srt instead of something.vtt, as I originally suggested.

**Reply by imtrobin** [reply]
I replaced them with XXX,YYY for privacy. I didn't have time to put them manually, I'm sure your method will work, thanks. I was hoping the tool would do automated as I have a lot of videos.

**Reply by GrampaWildWilly** [reply]
Yes, I figured the XXXXX stuff was the words of the subtitles & you quite properly wanted to keep that private.  But the YYYYY stuff isn't private, is it?  It's just sequence numbers, isn't it?

**Reply by imtrobin** [reply]
yup
